apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaConnectionStatus.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.kafka;
+
+enum KafkaConnectionStatus {
+    CONNECTED, UNCONNECTED
+}
```

Comment:
@darcydai
Hi, I don't found the retrying machine. I mean when is the status set to UNCONNECTED?

Reply:
yes. firstly,  UNCONNECTED should named DISCONNECTED, it is a bad spell,  I already update it in the last commit.
actually, DISCONNECTED is never used, because KafkaProducer has no way to measure the Kafka producer status in its API, maybe it will have the way in later.  Reconnection mechanism has implemented by KafkaProdcer if KafkaProdcer has been created, those works blew want to resolve KafkaProdcer create failed when the agent bootstrap

Reply:
I got it. That is why we didn't have this feature yet.